app/src/main/java/com/islaharper/dawnofandroid/presentation/components/TopBars.kt

```diff
@@ -0,0 +1,126 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.R
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AppTopBar(
+    title: String,
+    darkTheme: Boolean,
+    dynamicTheme: Boolean,
+    onSearchClicked: () -> Unit,
+    onThemeSwitchClicked: () -> Unit,
+    onDynamicSwitchClicked: () -> Unit
+) {
+    var expanded by remember {
+        mutableStateOf(false)
+    }
+
+    TopAppBar(
+        title = {
+            Text(
+                text = title,
+                color = MaterialTheme.colorScheme.onPrimary
+            )
+        },
+        actions = {
+            IconButton(onClick = onSearchClicked) {
+                Icon(
+                    Icons.Default.Search,
+                    contentDescription = stringResource(R.string.search_icon),
+                    tint = MaterialTheme.colorScheme.onPrimary
+                )
+            }
+            IconButton(onClick = {
+                expanded = true
+            }) {
+                Icon(
+                    imageVector = Icons.Default.MoreVert,
+                    contentDescription = "Open Options",
+                    tint = MaterialTheme.colorScheme.onPrimary
+                )
+            }
+            DropdownMenu(
+                modifier = Modifier.background(MaterialTheme.colorScheme.primaryContainer),
+                expanded = expanded,
+                onDismissRequest = {
+                    expanded = false
+                }
+            ) {
+                Text(
+                    text = stringResource(R.string.light_dark),
+                    modifier = Modifier.padding(start = 16.dp),
+                    color = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+                ThemeSwitcher(
+                    darkTheme = darkTheme,
+                    onClick = onThemeSwitchClicked
+                )
+
+                Text(
+                    text = stringResource(R.string.dynamic_theme),
+                    modifier = Modifier.padding(start = 16.dp),
+                    color = MaterialTheme.colorScheme.onPrimaryContainer
+                )
+                ThemeSwitcher(
+                    darkTheme = dynamicTheme,
+                    onClick = onDynamicSwitchClicked
+                )
+            }
+        },
+        colors = TopAppBarDefaults.smallTopAppBarColors(
+            containerColor = MaterialTheme.colorScheme.primary
+        )
+    )
+}
+
+@Composable
+fun SearchTopBar(
+    text: String,
+    onTextChanged: (String) -> Unit,
+    onSearchClicked: (String) -> Unit,
+    onCloseClicked: () -> Unit
+) {
+    SearchWidget(
```

Comment:
Where is this component? Is it one you have built yourself? I can't see it in this branch

Reply:
Thank you for adding this in, Is it possible to write a preview for this component?